streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -908,7 +912,7 @@ void runOnce() {
 
                 timeSinceLastPoll = Math.max(now - lastPollMs, 0);
 
-                if (maybePunctuate() || maybeCommit()) {
+                if (maybePunctuate() || (!eosEnabled && maybeCommit())) {
```

Comment:
Don't know if this is necessary, but it seemed to make sense that since we are checking inside the loop we don't need to check here as well.  But I'm open to removing this.